### PR TITLE
Fix droid repair

### PR DESCRIPTION
**Pull request by TiviPlus**
## Changelog
:cl:
fix: droids can be repaired again
/:cl:

### CODE REVIEW

File: code/modules/vehicles/unmanned/unmanned_droid.dm [hunk 1]
```diff
@@ -29,6 +29,33 @@
...
+/obj/vehicle/unmanned/droid/welder_act(mob/living/user, obj/item/I)
+	if(user.do_actions)
+		balloon_alert(user, "Already busy!")
+		return FALSE
+	if(obj_integrity >= max_integrity)
+		return TRUE
+	if(user.skills.getRating("engineer") < SKILL_ENGINEER_ENGI)
+		balloon_alert_to_viewers("[user] fumbles repairs", ignored_mobs = user)
+		balloon_alert(user, "You fumble repair")
+		var/fumbling_time = 10 SECONDS - 2 SECONDS * user.skills.getRating("engineer")
```
Comment (BraveMole) [inline code/modules/vehicles/unmanned/unmanned_droid.dm:41]: useless var, don't think that add that much readabiliy

File: code/modules/vehicles/unmanned/unmanned_droid.dm [hunk 2]
```diff
@@ -29,6 +29,33 @@
...
+		if(!do_after(user, fumbling_time, TRUE, src, BUSY_ICON_UNSKILLED, extra_checks = CALLBACK(I, /obj/item/tool/weldingtool.proc/isOn)))
+			return FALSE
+	balloon_alert_to_viewers("[user] starts repairs", ignored_mobs = user)
+	balloon_alert(user, "You start repair")
+	if(!do_after(user, 2 SECONDS, extra_checks = CALLBACK(I, /obj/item/tool/weldingtool.proc/isOn)))
+		balloon_alert_to_viewers("Stops repair")
```
Comment (BraveMole) [inline code/modules/vehicles/unmanned/unmanned_droid.dm:47]: stops repairing, stops repairs?

File: code/modules/vehicles/unmanned/unmanned_droid.dm [hunk 3]
```diff
@@ -29,6 +29,33 @@
...
+		return
+	if(!I.use_tool(src, user, 0, volume=50, amount=1))
+		return TRUE
+	obj_integrity += min(10, max_integrity-obj_integrity)
+	if(obj_integrity == max_integrity)
+		balloon_alert_to_viewers("Fully repaired!")
+	else
+		balloon_alert_to_viewers("[user] repairs", ignored_mobs = user)
+		balloon_alert(user, "You repair damage")
```
Comment (BraveMole) [inline code/modules/vehicles/unmanned/unmanned_droid.dm:56]: damages?

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/modules/vehicles/unmanned/unmanned_droid.dm (hunk 1 above)
Comment (TiviPlus) [reply code/modules/vehicles/unmanned/unmanned_droid.dm:41]: it makes the line stupid long this is fine imo

File: code/modules/vehicles/unmanned/unmanned_droid.dm (hunk 2 above)
Comment (TiviPlus) [reply code/modules/vehicles/unmanned/unmanned_droid.dm:47]: changed

File: code/modules/vehicles/unmanned/unmanned_droid.dm (hunk 3 above)
Comment (TiviPlus) [reply code/modules/vehicles/unmanned/unmanned_droid.dm:56]: damages isnt the correct term here as damages = payment for damage

**Review by BraveMole — APPROVED** [review]

**Pull request merged**